aws/resource_aws_guardduty_filter.go

```diff
@@ -0,0 +1,493 @@
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"action": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"NOOP",
+					"ARCHIVE",
+				}, false),
+			},
+			"rank": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Second),
+			Update: schema.DefaultTimeout(60 * time.Second),
+		},
+	}
+}
+
+func resourceAwsGuardDutyFilterCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	input := guardduty.CreateFilterInput{
+		Action:      aws.String(d.Get("action").(string)),
+		Description: aws.String(d.Get("description").(string)),
+		DetectorId:  aws.String(d.Get("detector_id").(string)),
+		Name:        aws.String(d.Get("name").(string)),
+		Rank:        aws.Int64(int64(d.Get("rank").(int))),
+	}
+
+	// building `FindingCriteria`
+	findingCriteria := d.Get("finding_criteria").([]interface{})[0].(map[string]interface{})
+
+	var err error
+	input.FindingCriteria, err = serializeFindingCriteria(findingCriteria)
+	if err != nil {
+		return err
+	}
+
+	tagsInterface := d.Get("tags").(map[string]interface{})
+	if len(tagsInterface) > 0 {
+		tags := make(map[string]*string, len(tagsInterface))
+		for i, v := range tagsInterface {
+			tags[i] = aws.String(v.(string))
+		}
+
+		input.Tags = tags
+	}
+
+	log.Printf("[DEBUG] Creating GuardDuty Filter: %s", input)
+	output, err := conn.CreateFilter(&input)
+	if err != nil {
+		return fmt.Errorf("Creating GuardDuty Filter %s failed: %s", input, err.Error())
+	}
+
+	d.SetId(strings.Join([]string{d.Get("detector_id").(string), *output.Name}, "_"))
+
+	return resourceAwsGuardDutyFilterRead(d, meta)
+}
+
+func resourceAwsGuardDutyFilterRead(d *schema.ResourceData, meta interface{}) error {
+	var detectorId, name string
+	var err error
+
+	if _, ok := d.GetOk("detector_id"); !ok {
+		// If there is no "detector_id" passed, then it's an import.
+		detectorId, name, err = parseImportedId(d.Id())
+		if err != nil {
+			return err
+		}
+	} else {
+		detectorId = d.Get("detector_id").(string)
+		name = d.Get("name").(string)
+	}
+
+	conn := meta.(*AWSClient).guarddutyconn
+
+	input := guardduty.GetFilterInput{
+		DetectorId: aws.String(detectorId),
+		FilterName: aws.String(name),
+	}
+
+	log.Printf("[DEBUG] Reading GuardDuty Filter: %s", input)
+	filter, err := conn.GetFilter(&input)
+
+	if err != nil {
+		if isAWSErr(err, guardduty.ErrCodeBadRequestException, "The request is rejected because the input detectorId is not owned by the current account.") {
+			log.Printf("[WARN] GuardDuty detector %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Reading GuardDuty Filter '%s' failed: %s", name, err.Error())
```

Comment:
Please update the `fmt.Errorf` calls to return the error object itself using the `%w` directive